Add disk usage info to '{Start,Secure} Cluster'

rmloveland opened the pull request:
Fixes #2569

Specifically:

1. This is listed as a known limitation of v1.1 and in v2.0, so we break
   that content out into an include

2. From there, we use that include in the various places that make note
   of that limitation (basically, the existing Admin UI docs)

3. Then, we can easily *add* the information to the various 'Start/Secure
   a Cluster' pages by snarfing the include

Note: I tried, but was unable to add the included note to the "Start a
Cluster in Docker" pages due to
https://github.com/cockroachdb/docs/issues/2659

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/docs/2660)
<!-- Reviewable:end -->

jseldess commented on _includes/available-capacity-metric.md:
```diff
@@ -0,0 +1 @@
+If you are running multiple nodes on a single machine (not recommended in production) and didn't specify the maximum allocated storage capacity for each node using the [`--store`](start-a-node.html#store) flag, the available capacity shown in the [**Capacity**](admin-ui-storage-dashboard.html#capacity) graph in the Admin UI is incorrect. This is because when multiple nodes are running on a single ...
```
We should probably expand this to also reference **Capacity Used** in the **Summary** panel:

```
... the **Capacity Used** metric in the [**Summary**](admin-ui-access-and-navigate.html#summary-panel) panel and the available capacity shown in the [**Capacity**](admin-ui-storage-dashboard.html#capacity) graph...
```

rmloveland replied:
Sweet, thanks!  Updated as suggested in bac307f9

jseldess commented on v1.1/secure-a-cluster.md:
```diff
@@ -212,6 +212,10 @@ As mentioned earlier, CockroachDB automatically replicates your data behind-the-
 
 The replica count on each node is identical, indicating that all data in the cluster was replicated 3 times (the default).
 
+{{site.data.alerts.callout_info}}
```
To keep these getting started pages lean, perhaps instead of embedding the include, we could link to the known limitation, for example:

```
{{site.data.alerts.callout_info}}Capacity metrics can be incorrect when running multiple nodes on a single machine. For more details, see this [limitation](known-limitations.html#available-capacity-metric-in-the-admin-ui).{{site.data.alerts.end}}
```

rmloveland replied:
Great suggestion - updated in 3e80ace3

jseldess commented on v1.1/start-a-local-cluster.md:
```diff
@@ -181,6 +181,10 @@ As mentioned earlier, CockroachDB automatically replicates your data behind-the-
 
 The replica count on each node is identical, indicating that all data in the cluster was replicated 3 times (the default).
 
+{{site.data.alerts.callout_info}}
```
Same as comment above:

To keep these getting started pages lean, perhaps instead of embedding the include, we could link to the known limitation, for example:

```
{{site.data.alerts.callout_info}}Capacity metrics can be incorrect when running multiple nodes on a single machine. For more details, see this [limitation](known-limitations.html#available-capacity-metric-in-the-admin-ui).{{site.data.alerts.end}}
```

rmloveland replied:
Updated in 3e80ace3

jseldess commented on v2.0/admin-ui-access-and-navigate.md:
```diff
@@ -56,7 +56,7 @@ Metric | Description
 Total Nodes | The total number of nodes in the cluster. <a href='admin-ui-access-and-navigate.html#decommissioned-nodes'>Decommissioned nodes</a> are not included in the Total Nodes count. <br><br>You can further drill down into the nodes details by clicking on [**View nodes list**](#nodes-list).
 Dead Nodes | The number of [dead nodes](admin-ui-access-and-navigate.html#dead-nodes) in the cluster.
-Capacity Used | The storage capacity used as a percentage of total storage capacity allocated across all nodes.<br><br>**Note:** If you are running multiple nodes on a single machine (not recommended), this value may be incorrect. This is because when multiple nodes are running on a single machine, the machine's hard disk is treated as separate stores for each node, while in reality, only one har ...
+Capacity Used | The storage capacity used as a percentage of total storage capacity allocated across all nodes.<br /><br />**Note**: {% include available-capacity-metric.md %}
```
It looks like using an include in the table does, in fact, break things: http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/ae5878f35d2552936f680b7306a56385dee2ebbf/stable/admin-ui-access-and-navigate.html#summary-panel

Can you play around with it and, if necessary, paste the text here directly?

jseldess replied:
You could also just do the include after the table, as you've done in other places.

rmloveland replied:
Thanks for the catch!  I updated this page and the v1.1 version to do the include after the table in 7076917f.  Tables appear to be building correctly now "on my machine (tm)" :-)

jseldess commented on v2.0/secure-a-cluster.md:
```diff
@@ -212,6 +212,10 @@ As mentioned earlier, CockroachDB automatically replicates your data behind-the-
 
 The replica count on each node is identical, indicating that all data in the cluster was replicated 3 times (the default).
 
+{{site.data.alerts.callout_info}}
```
Same as for v1.1 version:

To keep these getting started pages lean, perhaps instead of embedding the include, we could link to the known limitation, for example:

```
{{site.data.alerts.callout_info}}Capacity metrics can be incorrect when running multiple nodes on a single machine. For more details, see this [limitation](known-limitations.html#available-capacity-metric-in-the-admin-ui).{{site.data.alerts.end}}
```

rmloveland replied:
Thanks, addressed in eacc8ede

jseldess commented on v2.0/start-a-local-cluster.md:
```diff
@@ -181,6 +181,10 @@ As mentioned earlier, CockroachDB automatically replicates your data behind-the-
 
 The replica count on each node is identical, indicating that all data in the cluster was replicated 3 times (the default).
 
+{{site.data.alerts.callout_info}}
```
Same as for v1.1 version:

To keep these getting started pages lean, perhaps instead of embedding the include, we could link to the known limitation, for example:

```
{{site.data.alerts.callout_info}}Capacity metrics can be incorrect when running multiple nodes on a single machine. For more details, see this [limitation](known-limitations.html#available-capacity-metric-in-the-admin-ui).{{site.data.alerts.end}}
```

rmloveland replied:
Addressed in eacc8ede

jseldess commented:
Also, you need to update your copy of master, rebase this branch against it, and then push again to pass TC:

```
git checkout master
git pull master
git checkout fixitday-local-cluster-disk-usage
git rebase master
git push --force origin fixitday-local-cluster-disk-usage
```

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/d23267e5fca8cc734bee402e48d8d8d95f638bdc/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/bac307f9f6d791f7fe7481319a733643aee8004b/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/3e80ace3167360e8ef904bf33c24ffad6dada7f3/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/eacc8ede9d64d732df464a65a03f31043f6f2d97/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/7076917f5ee7d9fefa58a25143497643c5131ac3/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/cbf62dceaaf498ca77c88523cf859506b561e7f7/

rmloveland commented:
@jseldess I have a couple of questions:

- Do you have further feedback?  I think I've addressed the issues you pointed out, but I'm happy to fix any other things you noticed

- Should I ask for another review or is that not necessary since we are exposing information that already existed elsewhere in the docs, just in more places?

jseldess commented:
LGTM, @rmloveland! 

Please add `Fixes #<issue number>` to your PR description so the related issue auto-resolves. And then merge away!
